actions/security.go

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package actions
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/eclipse/codewind-installer/utils"
+	"github.com/eclipse/codewind-installer/utils/security"
+	"github.com/urfave/cli"
+)
+
+// SecurityTokenGet : Authenticate and retrieve an access_token
+func SecurityTokenGet(c *cli.Context) {
+	auth, err := security.SecAuthenticate(c, "", "")
+	if err == nil && auth != nil {
+		utils.PrettyPrintJSON(auth)
+	} else {
+		fmt.Println(err.Error())
+	}
+	os.Exit(0)
+}
+
+// SecurityCreateRealm : Create a realm in Keycloak
+func SecurityCreateRealm(c *cli.Context) {
+	err := security.SecRealmCreate(c)
+	if err != nil {
+		fmt.Println(err.Error())
+	} else {
+		utils.PrettyPrintJSON(security.Result{Status: "OK"})
+	}
+	os.Exit(0)
+}
+
+// SecurityClientCreate : Create a new client in Keycloak
+func SecurityClientCreate(c *cli.Context) {
+	err := security.SecClientCreate(c)
+	if err != nil {
+		fmt.Println(err.Error())
+	} else {
+		utils.PrettyPrintJSON(security.Result{Status: "OK"})
+	}
+	os.Exit(0)
+}
+
+// SecurityClientGet : Retrieve a client configuration from Keycloak
+func SecurityClientGet(c *cli.Context) {
+	registeredClient, err := security.SecClientGet(c)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(0)
+	}
+	if registeredClient != nil {
+		utils.PrettyPrintJSON(registeredClient)
+		os.Exit(0)
+	}
+	utils.PrettyPrintJSON(security.Result{Status: "Not found"})
+	os.Exit(0)
```

Comment:
os.Exit(0) is probably the wrong way to end the commands.
- log.Fatal(msg) seems to be what we do elsewhere.
- The editor plugins look at the return codes to see if there's a problem. Having os.Exit(0) on an error path is probably going to cause them problems.

Reply:
I'll check with the IDE folks to see what they are expecting. My thought process was,  the Exit(0) is that the CLI command completed correctly with no fault.  However, that does not mean the result of the command was successful if the backend server the CLI is talking to returns an error message.  In that last case the CLI would return a JSON response with the addition of {error: "operation", "error-description": "some meaningful text"}.   That message is still output and returned by the CLI for a human or the IDE to read.